### PR TITLE
implemented solution for advanced oop task

### PR DESCRIPTION
**Pull request by olieniev**

### CODE REVIEW

File: src/main/java/core/basesyntax/Main.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        for (int i = 0; i < figureArraySize; i++) {
+            if (i < figureArraySize / 2) {
+                figureArray[i] = figureSupplier.getRandomFigure();
+            } else {
+                figureArray[i] = figureSupplier.getDefaultFigure();
+            }
+        }
+        for (int j = 0; j < figureArraySize; j++) {
+            figureArray[j].draw();
+        }
```
Comment (Sanderfuck) [inline src/main/java/core/basesyntax/Main.java:18]: For what you need one more loop? You can make it using only one

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            case 2 -> new RightTriangle(random.nextDouble(), random.nextDouble(),
+                    colorSupplier.getRandomColor().name());
+            case 3 -> new Circle(random.nextDouble(), colorSupplier.getRandomColor().name());
+            default -> new IsoscelesTrapezoid(random.nextDouble(),random.nextDouble(),
+                    random.nextDouble(), colorSupplier.getRandomColor().name());
+        };
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_RADIUS,Color.WHITE.name());
```
Comment (Sanderfuck) [inline src/main/java/core/basesyntax/FigureSupplier.java:26]: Check style here

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        int randomFigureNumber = random.nextInt(NUMBER_OF_FIGURES);
+        return switch (randomFigureNumber) {
+            case 0 -> new Square(random.nextDouble(), colorSupplier.getRandomColor().name());
+            case 1 -> new Rectangle(random.nextDouble(), random.nextDouble(),
+                    colorSupplier.getRandomColor().name());
+            case 2 -> new RightTriangle(random.nextDouble(), random.nextDouble(),
+                    colorSupplier.getRandomColor().name());
+            case 3 -> new Circle(random.nextDouble(), colorSupplier.getRandomColor().name());
+            default -> new IsoscelesTrapezoid(random.nextDouble(),random.nextDouble(),
+                    random.nextDouble(), colorSupplier.getRandomColor().name());
```
Comment (Sanderfuck) [inline src/main/java/core/basesyntax/FigureSupplier.java:21]: See duplication code, you can create private methods and reuse them

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public Figure getRandomFigure() {
+        int randomFigureNumber = random.nextInt(NUMBER_OF_FIGURES);
+        return switch (randomFigureNumber) {
+            case 0 -> new Square(random.nextDouble(), colorSupplier.getRandomColor());
+            case 1 -> new Rectangle(random.nextDouble(), random.nextDouble(),
+                    colorSupplier.getRandomColor());
+            case 2 -> new RightTriangle(random.nextDouble(), random.nextDouble(),
+                    colorSupplier.getRandomColor());
+            case 3 -> new Circle(random.nextDouble(), colorSupplier.getRandomColor());
+            default -> new IsoscelesTrapezoid(random.nextDouble(),random.nextDouble(),
```
[the remark this answers is not in the document]
Comment (Sanderfuck) [reply src/main/java/core/basesyntax/FigureSupplier.java:20]: Check style

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Main.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        Figure[] figureArray = new Figure[figureArraySize];
+
+        for (int i = 0; i < figureArraySize; i++) {
+            if (i < figureArraySize / 2) {
+                figureArray[i] = figureSupplier.getRandomFigure();
+            } else {
+                figureArray[i] = figureSupplier.getDefaultFigure();
+            }
+            figureArray[i].draw();
+        }
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Main.java:16]: why do we need array? our task is just to create figures, we dont need to use them in future to store them in array for that
```suggestion
        for (int i = 0; i < figureArraySize; i++) {
            
            if (i < figureArraySize / 2) {
                figureSupplier.getRandomFigure().draw();
            } else {
                figureSupplier.getDefaultFigure().draw();
            }
        }
```

**Review by kerrrusha — APPROVED** [review]

**Pull request closed without merging**